### PR TITLE
Change user cancelled behavior in custom entitlement computation mode

### PR DESCRIPTION
**Pull request by tonidero**
### Description
Change user cancelled behavior in custom entitlement mode. Now for

### CODE REVIEW

**Comment by NachoSoto** [comment]
TODO: add an assertion to ensure there's no customer info request in this case.

File: Sources/Purchasing/Purchases/PurchasesOrchestrator.swift [hunk 1]
```diff
@@ -449,6 +451,9 @@ final class PurchasesOrchestrator {
                     return try await sk2Product.purchase(options: options)
                 }
         } catch StoreKitError.userCancelled {
+            if self.systemInfo.dangerousSettings.customEntitlementComputation {
+                throw ErrorUtils.purchaseCancelledError()
+            }
```
Comment (tonidero) [inline Sources/Purchasing/Purchases/PurchasesOrchestrator.swift:456]: I'm not sure if this error would be catched by the following catches... Still had to test each of these cases to make sure it behaved as I expected

File: Sources/Purchasing/Purchases/PurchasesOrchestrator.swift [hunk 2]
```diff
@@ -413,9 +413,11 @@ final class PurchasesOrchestrator {
                     productIdentifier: product.id,
                     error: error
                 ))
+                let publicError = ErrorUtils.purchasesError(withUntypedError: error).asPublicError
+                let userCancelled = publicError.code == ErrorCode.purchaseCancelledError.rawValue
```
Comment (NachoSoto) [inline Sources/Purchasing/Purchases/PurchasesOrchestrator.swift:417]: Oh I see why you needed this now. We need to compare the error domain too though, I'll fix that.

Comment (NachoSoto) [reply Sources/Purchasing/Purchases/PurchasesOrchestrator.swift:417]: I'm changing it to this:
```suggestion
                let userCancelled = publicError.isCancelledError
```

File: Sources/Purchasing/Purchases/PurchasesOrchestrator.swift [hunk 3]
```diff
@@ -466,6 +471,9 @@ final class PurchasesOrchestrator {
         // This detects if `Product.PurchaseResult.userCancelled` is true.
         let (userCancelled, sk2Transaction) = try await self.storeKit2TransactionListener
             .handle(purchaseResult: result)
+        if self.systemInfo.dangerousSettings.customEntitlementComputation {
```
Comment (NachoSoto) [inline Sources/Purchasing/Purchases/PurchasesOrchestrator.swift:474]: I think this was missing also checking `userCancelled`.

**Comment by NachoSoto** [comment]
Closing in favor of #2449

**Pull request closed without merging**